pkg/virt-controller/network/vmcontroller.go

```diff
@@ -0,0 +1,160 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"context"
+	"fmt"
+
+	k8sv1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	kubevirt "kubevirt.io/client-go/generated/kubevirt/clientset/versioned"
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+	"kubevirt.io/kubevirt/pkg/network/namescheme"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+)
+
+type clusterConfigChecker interface {
+	HotplugNetworkInterfacesEnabled() bool
+}
+
+type VMNetController struct {
+	clientset     kubevirt.Interface
```

Comment:
You may want to require only the `vmi` client to follow the narrow interface approach you are establishing here.

Reply:
I can try to narrow this down and see how it looks.
But can we do this as a follow up? I prefer to take this in to have a baseline for further work in this area.

Do we have an example of it used in a narrow manner like this in another place?

Reply:
I am ok to take this in follow up. Example can be found in `pkg/virt-handler/node-labeller/node_labeller.go`, I think we generally suffer from too broad interfaces :)

Reply:
ack
I'll make sure this will be followed up with.

@orelmisan , would you be able to help improve this as a follow up PR?

Reply:
Sure.